src/snowcli/cli/snowpark/function/commands.py

```diff
@@ -164,22 +105,30 @@ def function_update(
     replace: bool = ReplaceOption,
     **options,
 ) -> CommandResult:
-    """Updates an existing python UDF or UDTF using a local artifact."""
+    """Deploy a python UDF or UDTF using a local artifact."""
     snowpark_package(
         pypi_download,  # type: ignore[arg-type]
         check_anaconda_for_pypi_deps,
         package_native_libraries,  # type: ignore[arg-type]
     )
 
     fm = FunctionManager()
-    sm = StageManager()
+    function_exists = True
+    deploy = False
 
     try:
-        current_state = fm.describe(identifier)
-    except:
-        log.info(f"Function does not exists. Creating it from scratch.")
-        replace = True
-    else:
+        current_state = fm.describe(remove_parameter_names(identifier))
+    except ProgrammingError as ex:
+        if ex.msg.__contains__("does not exist or not authorized"):
```

Comment:
Maybe we can find other way for determining if a function exists? Maybe we can use `SHOW FUNCTIONS LIKE <pattern>` and then filter the results by matching arguments list?

Why?
1) IMO it's dangerous to base our logic on a message of exception from Snowflake.
2) "does not exist or not authorized" means an ambigous state - we don't know if the function really does not exist.

Reply:
Discussed and agreed that it's not so easy to use `show function like`. Let's stay with the current solution and maybe we'll revisit it in the future.